### ISSUE TITLE
onChange event not properly triggered when value is selected from autocomplete list

### ISSUE THREAD

**Issue by rernens** [udos86/ng-dynamic-forms#537]
## I'm submitting a...

<pre><code>
[x] Bug
[ ] Feature Request / Proposal
[ ] Question
</code></pre>

## I'm using...

<pre><code>
NG Dynamic Forms Version: 1.4.33

[ ] Basic UI
[ ] Bootstrap UI  
[ ] Foundation UI
[ ] Ionic UI
[ ] Kendo UI
[x] Material  
[ ] NG Bootstrap
[ ] Prime NG
</code></pre>

## Description

When used in conjunction with an autocomplete list, on change event is not triggered on a DynamicInputModel with the value selected from the autocomplete list but with the value entered in the field.

I am dynamically building the list array using a background service that listen to the typed value in the field 

```
let birthCityModel = this.formService.findById('birthCity', this.formModel) as DynamicInputModel;
    birthCityModel.valueUpdates
      .debounceTime(500)
      .distinctUntilChanged()
      .subscribe( ( value: string ) => this.openApi.openDataSoft('insee', value ).subscribe( data => birthCityModel.list = this.buildCitiesList( data )));

```

As at this point there is no way to store objects in the list attribute of a DynamicInputModel, I keep the lis in sync with an array of objects to be able to access an id associated with the value selected in the list.

```
  private buildCitiesList( data ) {

    let cities = [];
    this.cities = [];

    if ( data.records ) {
      for( let record of data.records ) {
        let city = {};
        let name = record.fields['nom_com'] + ' ( ' + record.fields['nom_dept'] + ' )';
        city['name'] = name;
        city['insee'] = record.fields['insee_com'];
        cities.push( name );
        this.cities.push( city );
      }
    }

    return cities;
  }
```

I am using the onChange event to extract from the array of object the relevant id for the named value selected in the autocomplete list.

When doing so, the value returned by the model.value is the value type in the field not the selected value from the autocomplete list.

```
public onFormChange( changes: SimpleChanges ) {
      let formModel = changes.model;
      let id = formModel['id'];
      let value = formModel['value'];
      this.log.debug(value); // value shown is the typed value
      switch ( id ) {
        case 'birthCity':
          this.log.debug(this.cities);
          let cities = this.cities.filter( city =>  city.name === value ) ;
          if ( cities.length === 1 )
            this.checkSSN.insee = cities[0].insee;
          break;
        case 'birthCountry':
          if (value !== '33000')
            this.checkSSN.insee = value;
          break;
        default:
          break;
      }
  }
```
the onChange event is not triggered after the value selected in the autocomplete list replaces the entered data in the field while the updateValues method is well invoked and reflect the changes in the field.

MORE !!! Same applies to datepicker ...

selecting the date from the calendar pop-up leads to similar behavior : onChange event reflect the previous date contained in the field not the selected date from the popup.

**Comment by udos86** [comment]
@rernens 

There should alreadybe  a merged fix for this which will be released in the next patch version `1.4.34`.

**Comment by rernens** [comment]
@udos86 

Thanks Udo, looking for it.

**Comment by rernens** [comment]
@udos86 

fixed in 1.4.34 thanks.
